src/NumericalAlgorithms/Interpolation/Events/InterpolateWithoutInterpComponent.hpp

```diff
@@ -0,0 +1,193 @@
+      VolumeDim, InterpolationTargetTag, Metavariables, Tensors, RegistrarList>;
+};
+}  // namespace Registrars
+/// \endcond
+
+namespace Events {
+/// Does an interpolation onto an InterpolationTargetTag by calling Actions on
+/// the InterpolationTarget component.
+template <size_t VolumeDim, typename InterpolationTargetTag,
+          typename Metavariables, typename... Tensors, typename EventRegistrars>
+class InterpolateWithoutInterpComponent<VolumeDim, InterpolationTargetTag,
+                                        Metavariables, tmpl::list<Tensors...>,
+                                        EventRegistrars>
+    : public Event<EventRegistrars> {
+  /// \cond
+  explicit InterpolateWithoutInterpComponent(
+      CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(InterpolateWithoutInterpComponent);  // NOLINT
+  /// \endcond
+
+  using options = tmpl::list<>;
+  static constexpr OptionString help =
+      "Does interpolation using the given InterpolationTargetTag, "
+      "without an Interpolator ParallelComponent.";
+
+  static std::string name() noexcept {
+    return option_name<InterpolationTargetTag>();
+  }
+
+  InterpolateWithoutInterpComponent() = default;
+
+  using argument_tags =
+      tmpl::list<::Tags::TimeStepId, Tags::InterpPointInfo<Metavariables>,
+                 domain::Tags::Mesh<VolumeDim>, Tensors...>;
+
+  template <typename ParallelComponent>
+  void operator()(
+      const TimeStepId& time_id,
+      const db::item_type<Tags::InterpPointInfo<Metavariables>>& point_infos,
+      const Mesh<VolumeDim>& mesh,
+      const db::const_item_type<Tensors>&... tensors,
+      Parallel::ConstGlobalCache<Metavariables>& cache,
+      const ElementId<VolumeDim>& array_index,
+      const ParallelComponent* const /*meta*/) const noexcept {
+    // Get element logical coordinates of the target points.
+    const auto& block_logical_coords =
+        get<Vars::PointInfoTag<InterpolationTargetTag, VolumeDim>>(point_infos);
+    const std::vector<ElementId<VolumeDim>> element_ids{{array_index}};
+    const auto element_coord_holders =
+        element_logical_coordinates(element_ids, block_logical_coords);
+
+    // There is exactly one element_id in the list of element_ids.
+    if (element_coord_holders.count(element_ids[0]) == 0) {
+      // There are no target points in this element, so we don't need
+      // to do anything.
+      return;
+    }
+
+    // There are points in this element, so interpolate to them and
+    // send the interpolated data to the target.  This is done
+    // in several steps:
+    const auto& element_coord_holder = element_coord_holders.at(element_ids[0]);
+
+    // 1. Get the list of variables
+    Variables<typename InterpolationTargetTag::vars_to_interpolate_to_target>
+        interp_vars(mesh.number_of_grid_points());
+
+    // Clang-tidy wants extra braces for `if constexpr`
```

Comment:
Is this still true with clang-tidy 10?

Reply:
here too, clang-tidy works again.  Removing NOLINTs,,,